smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -134,6 +138,12 @@ contract MixinPurchase is
         address(unlockProtocol)
       );
     }
+
+    // get fee from Unlock 
+    uint protocolFee = (_keyPrice * unlockProtocol.fee()) / BASIS_POINTS_DEN;
+
+    // pay fee to Unlock
+    _transfer(tokenAddress, payable(address(unlockProtocol)), protocolFee);
```

Comment:
Wow, that's easy :D

Reply:
we have one issue though. Currently that fails with ERC20 because we receive all the ERC20 tokens in a single call at the end of the purchase loop BUT we trigger the `_recordKeyPurchase`  for each key within the loop. 

That means we attempt to transfer tokens that have not been received yet - which fails. 

It works fine with native tokens hat arrives directly to the contract via `payable` call.

The solution will be to take the protocol fee payment logic out of `_recordKeyPurchase` and call it once at the end of the purchase call. We already have a `_payReferrer` that could be reused for that maybe

Reply:
I was worried about contract size but seems to work fine after adding a `_payProtocol` internal func